infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -588,12 +605,43 @@ public void resetServer(@Nonnull RequestMsg req,
                 " sequencerEpoch = {}", globalLogTail, streamTailToGlobalTailMap.size(),
                 streamTailToGlobalTailMap, sequencerEpoch);
 
+        if (serverContext.isDeleteInactiveStreamsOnSequencerReset()) {
+            CompletableFuture<Set<UUID>> getInactiveStreamsFuture = CompletableFuture.supplyAsync(() ->
+                            getInactiveStreams(Collections.unmodifiableSet(streamsAddressMap.keySet())),
+                            corfuRuntimeExecutor)
+                    .exceptionally(ex -> {
+                        log.error("Failed to get inactive streams.", ex);
+                        return Collections.emptySet();});
+            getInactiveStreamsFuture.thenRunAsync(() -> deleteStreams(getInactiveStreamsFuture.join()),
+                            executor)
```

Comment:
I would suggest simplifying the threading model. Instead of executing one stage of the async flow on one executor and the other stage on another executor, I think its cleaner to just do it in a synchronous manner on on executor

Reply:
That introduces dead lock, doesn't it? Getting the inactive streams needs corfu runtime and table read on RegistryTable. That's why I unblock sequencer main thread here and put the table read on a different thread.

Reply:
Sorry, I think my comment wasn't clear, by

> synchronous manner on on executor

I meant why not just execute the logic on `corfuRuntimeExecutor`. Essentially, I don't understand why the workflow needs to be split across two executors.

Reply:
We know that this logic shouldn't block the main sequencer thread (deadlock), and the sequencer will serve other requests while the corfuRuntimeExecutor does the deleteStream. In that case, we need to synchronize between the two sequencer threads, which is even more complicated. That's why I think keeping the main sequencer thread the ONLY thread that modifies the maps is most straightforward.